kogito-codegen-modules/kogito-codegen-core/src/main/java/org/kie/kogito/codegen/core/io/CollectedResourceProducer.java

```diff
@@ -95,8 +103,8 @@ public static Collection<CollectedResource> fromJarFile(Path jarPath) {
      */
     public static Collection<CollectedResource> fromDirectory(Path path) {
         Collection<CollectedResource> resources = new ArrayList<>();
-        try (Stream<Path> paths = Files.walk(path)) {
-            paths.filter(Files::isRegularFile)
+        try (Stream<Path> paths = path.toFile().isHidden() ? Stream.empty() : Files.walk(path)) {
```

Comment:
I would probably do an if before the `try` so we can `log.debug` that the code is skipping a directory because hidden (it should help users in case of issues)
```suggestion
        if (path.toFile().isHidden()) {
            LOGGER.debug("Skipping directory because hidden: " + path);
            return resources;
        try (Stream<Path> paths = Files.walk(path)) {
```